internal/clients/interface.go

```diff
@@ -95,9 +98,9 @@ type Azure interface {
 
 	// CreateVM creates Azure virtual machine.
 	// Most of the parameters are constant for now.
-	// imageID is expected in format /subscriptions/{subscriptionID}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/images/{imageID}
-	// sshKeyBody should be full public key body
-	CreateVM(ctx context.Context, location string, resourceGroupName string, imageID string, pubkey *models.Pubkey, instanceType InstanceTypeName, vmName string) (*string, error)
+	// location - to deploy into
+	// imageName - the imageID will be inferred as /subscriptions/{subscriptionID}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/images/{imageName}
+	CreateVM(ctx context.Context, location string, resourceGroupName string, imageName string, pubkey *models.Pubkey, instanceType InstanceTypeName, vmName string) (*string, error)
 }
```

Comment:
Just a friendly reminder that Go doc syntax supports list, you must do it this way:

```
- a list
- another item
```

This will be formatted badly. Just saying, good for now as this will likely change: https://tip.golang.org/doc/comment